server/src/main/java/org/opensearch/extensions/action/ExtensionHandleTransportRequest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.action;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.transport.TransportRequest;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * This class encapsulates a transport request to extension
+ *
+ * @opensearch.api
```

Comment:
Not necessarily in this PR but someone should open an issue to discuss what extension classes should be api vs. internal.  I'm not so sure about this one.

Reply:
So this is a transport request going to an extension. The extension would use it as an API to understand the request. 
But sure let me open up and issue for this, I agree we need a little more wider discussion and alignment.

Reply:
Internals classes are termed as `internal` which can't be extended. Classes which can be extended are termed as `api`

Reply:
Agreed, but are we expecting users to extend this class or is it here for our convenience?

Reply:
Not really extend but we expect extensions to use it. Anyway I'll open up an issue for this. 
https://github.com/opensearch-project/opensearch-sdk-java/issues/168